src/uucore/src/lib/features/i18n/month.rs

```diff
@@ -0,0 +1,73 @@
+// This file is part of the uutils coreutils package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use std::sync::OnceLock;
+
+use icu_datetime::provider::neo::{DatetimeNamesMonthGregorianV1, MonthNames};
+use icu_locale::{Locale, locale};
+use icu_provider::prelude::*;
+
+use crate::i18n::get_time_locale;
+
+fn load_month_names(loc: &Locale) -> Option<Vec<(String, u8)>> {
+    let data_locale = DataLocale::from(loc.clone());
+    let abbr_attr = DataMarkerAttributes::from_str_or_panic("3");
+    let request = DataRequest {
+        id: DataIdentifierBorrowed::for_marker_attributes_and_locale(abbr_attr, &data_locale),
+        metadata: DataRequestMetadata::default(),
+    };
+
+    let response: DataResponse<DatetimeNamesMonthGregorianV1> =
+        icu_datetime::provider::Baked.load(request).ok()?;
+
+    if let MonthNames::Linear(names) = response.payload.get() {
+        let mut result = Vec::new();
+        for (i, name) in names.iter().take(12).enumerate() {
+            let month = (i + 1) as u8;
+            let upper = name.to_uppercase();
+            let stripped = upper.trim_end_matches('.');
+            if stripped != upper {
+                result.push((stripped.to_string(), month));
+            }
```

Comment:
The logic for handling periods is unclear - why only trim from the end?
maybe add a comment ?